### PR TITLE
Added a note about deprecating addClassesToCompile()

### LINKED ISSUE #7815: addClassesToCompile() has been deprecated
See https://github.com/symfony/symfony/pull/22459:

The `Extension::addClassesToCompile()` method has been deprecated and will be removed in 4.0.

### PR DESCRIPTION
**Pull request by javiereguiluz**
This fixes #7815.

### CODE REVIEW

File: bundles/extension.rst [hunk 1]
```diff
@@ -129,6 +129,11 @@ read more about it, see the ":doc:`/bundles/configuration`" article.
 Adding Classes to Compile
 -------------------------
 
+.. versionadded:: 3.3
+    This technique is discouraged and the ``addClassesToCompile()`` has been
```
Comment (xabbuh) [inline bundles/extension.rst]: I think we should reword this a bit as when #7193 is merged, we will also explain the `addAnnotatedClassesToCompile()` method in this section.

Comment (javiereguiluz) [reply bundles/extension.rst]: I don't know what to do here. What were you thinking exactly? Thanks!

**Review by xabbuh — APPROVED** [review]
I don't remember what I wanted to change regard my last comment.

**Comment by xabbuh** [comment]
Thank you Javier.

**Pull request closed without merging**